lib/views/scannerview.dart

```diff
@@ -30,122 +30,135 @@ class ScanScreenState extends State<ScanScreen> {
     super.dispose();
   }
 
-  Future<void> _startBarcodeScan() async {
+
+
+
+  Future<void> _startBarcodeScan(String barcodeScanRes) async {
     try {
-      String barcodeScanRes = await FlutterBarcodeScanner.scanBarcode(
-        '#ff6666',
-        'Cancel',
-        true,
-        ScanMode.BARCODE,
-      );
       if (barcodeScanRes != '-1') {
         setState(() {
           _scanResult = barcodeScanRes;
         });
 
-
         final updateOutTime = await _insertData.updateOuttime(barcodeScanRes);
         if (updateOutTime) {
           if (mounted) {
             ScaffoldMessenger.of(context).showSnackBar(
-                const SnackBar(content: Text('Out time updated'))
-            );
+                const SnackBar(content: Text('Out time updated')));
           }
-        }
-        else if(updateOutTime == false) {
-          final fetchFromArchiveStudent = await _insertData.fetchFromArchiveStudent(barcodeScanRes);
+        } else if (updateOutTime == false) {
+          final fetchFromArchiveStudent =
+          await _insertData.fetchFromArchiveStudent(barcodeScanRes); // TODO:can we remove this line?
           if (fetchFromArchiveStudent) {
-            Fluttertoast.showToast(
-              msg: 'Registered Successfully',
-              toastLength: Toast.LENGTH_SHORT,
-              gravity: ToastGravity.BOTTOM,
-              timeInSecForIosWeb: 5,
-              backgroundColor: Colors.black,
-              textColor: Colors.white,
-              fontSize: 16.0,
-            );
-          }
-          else {
-          final fetchStudent = await fetchStudentData(barcodeScanRes);
-          if (mounted) {
-            if (fetchStudent != null) {
-              Navigator.push(
-                context,
-                MaterialPageRoute(
-                  builder: (context) =>
-                      StudentView(
-                        rollno: fetchStudent.rollno,
-                        name: fetchStudent.name,
-                        department: fetchStudent.department,
-                      ),
-                ),
-              );
-            } else {
-              // Handle case where student data is not found
```

Comment:
scannerview : Remove navigation to student_detail_view.

As we are not taking signature of student we can remove the navigation